src/atomate2/common/analysis/defects/jobs.py

```diff
@@ -166,3 +182,183 @@ def get_ccd_documents(
+) -> Response:
+    """Perform charge defect supercell calculations.
+
+    Run a atomic relaxation calculation for each available charge state of the defect.
+
+    Parameters
+    ----------
+    defect:
+        A defect object representing the defect in a unit cell.
+    relax_maker:
+        A RelaxMaker object to use for the atomic relaxation.
+    sc_mat:
+        The supercell matrix. If None, the code will attempt to create a
+        nearly-cubic supercell.
+    defect_index:
+        Additional index to give unique names to the defect calculations.
+        Useful for external bookkeeping of symmetry distinct defects.
+    add_info:
+        Additional information to store with the defect cell relaxation calculation.
+        By default only the defect object and charge state are stored.
+    relaxed_sc_lattice:
+        The lattice of the relaxed supercell. If provided, the lattice parameters
+        of the supercell will be set to value specified.  Otherwise, the lattice it will
+        only by set by `defect.structure` and `sc_mat`.
+
+    Returns
+    -------
+    Response
+        A response object containing the summary of the calculations for different
+        charge states.
+    """
+    defect_q_jobs = []
+    all_chg_outputs = dict()
+    sc_def_struct = defect.get_supercell_structure(sc_mat=sc_mat)
+    sc_def_struct.lattice = relaxed_sc_lattice
+    if sc_mat is not None:
+        sc_mat = np.array(sc_mat).tolist()
+    for qq in defect.get_charge_states():
+        suffix = (
+            f" {defect.name} q={qq}"
+            if defect_index == ""
+            else f" {defect.name}-{defect_index} q={qq}"
+        )
+        charged_struct = sc_def_struct.copy()
+        charged_struct.set_charge(qq)
+        charged_relax = relax_maker.make(charged_struct)
+        charged_relax.append_name(suffix)
+
+        # write some provenances data in info.json file
+        info = {
+            "defect": defect,
+            "charge_state": qq,
+            "defect_name": defect.name,
+            "bulk_formula": defect.structure.composition.reduced_formula,
+            "bulk_num_sites": len(defect.structure),
+            "bulk_space_group_info": defect.structure.get_space_group_info(),
+            "sc_mat": sc_mat,
+        }
+
+        if add_info is not None:
+            info.update(add_info)
+        charged_relax.update_maker_kwargs(
+            {"_set": {"write_additional_data->info:json": info}}, dict_mod=True
+        )
+        defect_q_jobs.append(charged_relax)
+        charged_output: TaskDocument = charged_relax.output
+        all_chg_outputs[qq] = {
+            "structure": charged_output.structure,
+            "entry": charged_output.entry,
+            "dir_name": charged_output.dir_name,
+            "uuid": charged_relax.uuid,
+        }
+        # TODO: check that the charge state was set correctly
+
+    return Response(output=all_chg_outputs, replace=defect_q_jobs)
```

Comment:
Same with this here. E.g., see https://github.com/materialsproject/atomate2/blob/aa0bb0737eead988d1dc31fa46d0da0f4e61bc8c/src/atomate2/vasp/jobs/elastic.py#L226-L227

Reply:
You're right the outputs are all references. It's just that they are all meant to be comsumed by other jobs so this error did not affect anything.  I merged your change.